neo4j/internal/connector/connector.go

```diff
@@ -86,7 +86,7 @@ func (c Connector) Connect(address string, boltLogger log.BoltLogger) (db.Connec
 	}
 	config := tls.Config{InsecureSkipVerify: c.SkipVerify, RootCAs: c.RootCAs, ServerName: serverName}
 	tlsconn := tls.Client(conn, &config)
-	err = tlsconn.Handshake()
+	err = tlsconn.HandshakeContext(ctx)
```

Comment:
this is where I'm not sure... should I use another context for this specific TLS handshake or not?